game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/battlegroup/NormalBattleGroups.java

```diff
@@ -0,0 +1,159 @@
+package games.strategy.triplea.delegate.battle.steps.fire.battlegroup;
+
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.Side.OFFENSE;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.FIRST_STRIKE_UNITS;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.UNITS;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.data.properties.GameProperties;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.experimental.UtilityClass;
+import org.triplea.java.PredicateBuilder;
+
+/**
+ * Creates battle groups for units that use attack/defense attributes.
+ *
+ * <p>There is a FirstStrike battle group and a General battle group.
+ */
+@UtilityClass
+class NormalBattleGroups {
+
+  /**
+   * Creates battle groups for the first strike phase
+   *
+   * <p>If an opposing isDestroyer unit is present, then the phase for that side will not occur (see
+   * {@link #noDestroyerPresent} for both firing squadrons)
+   */
+  static BattleGroup createFirstStrike(
+      final Collection<UnitType> unitTypes, final GameProperties properties) {
+
+    final List<FiringSquadron> firingSquadrons =
```

Comment:
nit, var name perhaps could be more descriptive, this is the 'first strike' firing squadron?

Is this also the right way to group the units? I wonder if we should instead be going by unit-type to assign them to a phase rather than doing it per-phase. The latter can be complicated as we negate units and then add them later, which leads to logic seemingly in multiple places (ie: if this predicate is false here, it needs to be true in another place).

Reply:
I think your comment in the forum is similar to this idea you have here.  And it sounds fine except that the `FiringSquadron` needs to be able to handle existing logic as well.  So, the existing logic would have the more complicated logic while what ever is defined through the XML would have a much simpler logic.  But both would be expressed internally through the `FiringSquadron`.

Reply:
The question is whether which of the following two make more sense:
- group units into squadrons by phase
- group units into squadron by unit type

In other words, do we create squadrons first by looking at our unit types and then knowing which phases we would have, or do we start by knowing the full set of phases and then adding units when we get to the logic for each phase.

I do see here an issue where it is assumed that based on some conditions present in this class, that there are related conditions in another class. For example, if a unit is not included here, then we would presume they would be looped in at another phase. This seemingly does create a situation where we could have units not getting assigned to any phase just due to complexity and a logic bug.  I think this can be helped if we consolidate the logic for assigning units to phases, then we won't have a situation where we have different bits of code in different places doing essentially the same thing (DRY).